crates/trigger-http/src/lib.rs

```diff
@@ -456,10 +471,76 @@ pub(crate) trait HttpExecutor: Clone + Send + Sync + 'static {
 #[derive(Default)]
 pub struct HttpRuntimeData {
     origin: Option<String>,
+    engine: Option<Arc<TriggerAppEngine<HttpTrigger>>>,
+    handler: Option<HttpHandlerExecutor>,
     /// The hosts this app is allowed to make outbound requests to
     allowed_hosts: AllowedHostsConfig,
 }
 
+impl HttpRuntimeData {
+    fn chain_request(
+        data: &mut spin_core::Data<Self>,
+        request: wasmtime_wasi_http::types::OutgoingRequest,
+        component_id: String,
+    ) -> wasmtime::Result<
+        wasmtime::component::Resource<wasmtime_wasi_http::types::HostFutureIncomingResponse>,
+    >
+    where
+        Self: Sized,
```

Comment:
You can remove the `where Self: Sized` here.

Reply:
Copypasta, thanks for the spot!